Updated to work with Craft 5

johnfmorton opened the pull request:
I have updated the plugin for Craft 5. It maintains its compatibility with Craft 4.

Here are my change notes, which you can find in the `CHANGELOG.md` file.

## 5.0.0 - 2024-04-04
- Craft 5 support.
- Adjusted version number to reflect Craft 5 compatibility.
- Styling updates for the sidebar for User Manual content. Nested lists now have a different style to make them easier to read.
- Added conditional messaging to help provide instructions to the developer when the plugin is installed in a fresh installation of Craft CMS with no sections or entries yet created.
- Added craftMajorVersion() twig function to help style nested lists in Craft 4.

Also, the `README.md` file has instructions on installing the plugin from my GitHub repo and the `customization` branch. If you accept this pull request, please remove those instructions from the `README.md` file. 

Thanks! -John

johnfmorton committed:
feat: add structure page level indicator to admin index (Dalton Rooney)

johnfmorton committed:
icon for sub menu nav items

johnfmorton committed:
Fix: no longer hides global header

johnfmorton committed:
Fix: do not constrain #content area

johnfmorton committed:
Tweak the sub-menu icon and its placement

johnfmorton committed:
Updated version number to 5.0.0 - additional style tweaks

johnfmorton committed:
fixed - remove unneeded comma

johnfmorton committed:
note on using the WIP repo

johnfmorton committed:
Branch section retrieval for Craft 4 and Craft 5

johnfmorton committed:
Removed dupe line

johnfmorton committed:
Abstracted Craft version logic

johnfmorton committed:
reworked nesting styles

johnfmorton committed:
Added return types

johnfmorton committed:
Added warnings for incomplete set up of plugin

johnfmorton committed:
Added placeholder for plugin name override

johnfmorton committed:
Added style update for Craft 4 nested list

johnfmorton committed:
Updated for Craft 5 with note about GitHub installation

johnfmorton committed:
Updated the date

johnfmorton committed:
Language tweak

RobErskine commented:
Sorry, didn't get a chance to look at this over the weekend. Going to carve out some time after the eclipse this afternoon to get this merged in.

johnfmorton commented:
No problem. I'm happy to go over it with you on a call if you want. Just let me know.

RobErskine committed:
Preparing README for Craft 5 Release

- Removed support documentation for PR testing
- Cleaned up README styles
- Updated Release section

RobErskine committed:
Delete resources/sub-menu-icon.ai

- Removing source file. SVG should be fine.

RobErskine committed:
Update UserManual.css

- Updating code comments

RobErskine committed:
Update UserManualTwigExtension.php

- Fixing typo in the `!($sectionID)` warning

RobErskine reviewed (approved):
LGTM! Thanks @johnfmorton
